funcx_endpoint/tests/test_rabbit_mq/conftest.py

```diff
@@ -0,0 +1,22 @@
+import pika
+import pytest
+
+from funcx_endpoint.endpoint.rabbit_mq import ResultQueuePublisher
+
+CONN_PARAMS = pika.URLParameters("amqp://guest:guest@localhost:5672/")
+
+
+@pytest.fixture(scope="module")
+def ensure_result_queue():
+    result_pub = ResultQueuePublisher(
```

Comment:
I think trying to use this wrapper class to setup the connection during tests is a mistake. The class expects the queue to already exist, so I don't see this working out well.

I'd recommend just changing to
```python
conn = pika.BlockingConnection(CONN_PARAMS)
channel = conn.channel()
channel.confirm_delivery()
channel.exchange_declare(
    exchange=ResultQueuePublisher.EXCHANGE_NAME,
    exchange_type=ResultQueuePublisher.EXCHANGE_TYPE,
)
channel.queue_declare(queue=ResultQueuePublisher.QUEUE_NAME)
```

I'm not sure if a couple of those steps are needed. I'd have to lookup docs to see exactly.

Reply:
Yeah, this is now fixed. I've basically written out an explicit call to pika to create an exchange, queue and bind the queue to the routing_key.